### PR TITLE
Issue/58/ultra super class

### PR DESCRIPTION
**Pull request by matthewkirby**
Developed a super mega ultra class for every class outside of core to inherit. Contains an attribute that the manager class will read.

### CODE REVIEW

File: tests/test_clmmbase.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    class_list = list(set(class_list) - set(skip_classes) - set(extern_classes))
+
+    # Print the list of classes being analyzed. You will only see output
+    # on assertion failures
+    print(class_list)
+
+    # Assert that we have not added a new class to the api. If this fails, we
+    # need to instantiate an example of these objects below and test that
+    # ask_type has been set to a reasonable value
+    assert len(class_list) == 2
```
Comment (cavestruz) [inline tests/test_clmmbase.py]: This should not have to be modified with each new class.

File: tests/test_clmmbase.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    t1 = clmm.CLMMBase()
+    t1.ask_type = ['something']
+    np.testing.assert_array_equal(t1.ask_type, ['something'])
+
+    t2 = clmm.CLMMBase()
+    np.testing.assert_raises(TypeError, t2.ask_type, 3.14159)
+
+
+def test_new_classes():
```
Comment (cavestruz) [inline tests/test_clmmbase.py]: I like this check, but should be more automatic, and lets people know if they have to inherit this.

Comment (matthewkirby) [reply tests/test_clmmbase.py]: done

**Comment by matthewkirby** [comment]
I rewrote the tests to be automatic. Let me know if there are more changes that are requested.

**Pull request merged**